app/src/behaviors/behavior_combo.c

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_combo
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <sys/dlist.h>
+#include <kernel.h>
+
+#include <zmk/behavior.h>
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+//#if DT_NODE_EXISTS(DT_DRV_INST(0))
+
+// todo: max_keys could be set to the actual number of keys on the board
+#define ZMK_BHV_COMBO_MAX_POSITIONS 100
+
+// todo: move constants to a setting somewhere?
```

Comment:
Maybe should be Kconfig settings so you can adjust these if you want bigger combos?

Reply:
good suggestion, implemented.